kube-runtime/src/utils/event_flatten.rs

```diff
@@ -0,0 +1,122 @@
+use crate::watcher::{Error, Event};
+use core::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use futures::{Stream, TryStream};
+use pin_project::pin_project;
+
+// grab from private part of tokio
+macro_rules! ready {
```

Comment:
We already pull this in from `futures`: https://docs.rs/futures/latest/futures/macro.ready.html

Reply:
ah, lol, didn't realize they were the same.